.github/workflows/deploy_AppStore.yml

```diff
@@ -69,6 +69,13 @@ jobs:
       env:
         MATCH_PASSWORD: ${{ secrets.MATCH_PASS }}
 
+    # Fix issue with https://github.com/fastlane/fastlane/issues/20553.
+    - name: Set Up App Store API Secret  
+      env:
+        APP_STORE_KEY: ${{ secrets.APPSTORE_CONNECT_API_KEY }}
+      run: |
+        echo "$APP_STORE_KEY" > appstorekey.p8
```

Comment:
Based on [this practice](https://docs.github.com/en/actions/security-guides/encrypted-secrets#limits-for-secrets), I would prefer to use `gpg` and `the AES256 cipher algorithm` or `base64`. What do you think?

Reply:
@blyscuit Another way to fix this is using `distribution_manager.upload_to_testflight` instead. It will use `fastlane.pilot` instead of `fastlane.deliver`

Reply:
The current issue if with `fastlane. app_store_connect_api_key`. This PR might be best to wait until the error is fixed. I will retry with a different approach for GH Secrets but since `fastlane.app_store_connect_api_key` didn't work locally, I have some concern. 
The practices outlined in your link shows using file pushed to the repo and decrypted with secret, it's different than what we are doing here, which is saving the secret to a file. Could you explain you concern more on this one?

Reply:
@blyscuit 
> This PR might be best to wait until the error is fixed.

Yes, we could wait. However, after some research, the issue is more related to `fastlane.deliver`. We could use `fastlane.pilot` instead (a temporary fix)

> The practices outlined in your link shows using file pushed to the repo and decrypted with secret, it's different than what we are doing here, which is saving the secret to a file.

As I mentioned, it is a practice that we could follow. For example, with [base 64 aprroach](https://docs.github.com/en/actions/security-guides/encrypted-secrets#storing-base64-binary-blobs-as-secrets) the approach is:
1. Encode the file or the key with base64 method
2. Add the base 64 encoded string to secrets
3. In the CI workflow, decode the secret and `echo` it to a file (appstorkey.p8)

Technically, it's the same but with GitHub's practice, we will do extra works: encode and decode the secret with base64 method